src/borg/cache.py

```diff
@@ -435,10 +434,8 @@ def fetch_and_build_idx(archive_id, repository, key):
                     if not isinstance(item, dict):
                         logger.error('Error: Did not get expected metadata dict - archive corrupted!')
                         continue
-                    item = Item(internal_dict=item)
-                    if 'chunks' in item:
-                        for chunk_id, size, csize in item.chunks:
-                            chunk_idx.add(chunk_id, 1, size, csize)
+                    for chunk_id, size, csize in item.get(b'chunks', []):
+                        chunk_idx.add(chunk_id, 1, size, csize)
```

Comment:
These are both a bit older, this change was looked at previously in #1729 / https://github.com/borgbackup/borg/issues/235#issuecomment-254010549